linera-base/src/crypto.rs

```diff
@@ -23,7 +23,7 @@ use {
 };
 
 /// A signature key-pair.
-pub struct KeyPair(dalek::Keypair);
+pub struct KeyPair(dalek::SigningKey);
```

Comment:
Interesting

Reply:
I wonder if it's generally true for asymmetric signature algorithms that you can extract the verifying key from the signing key.

Reply:
I can't find an example where it wouldn't be the case, but the "extraction" might require arithmetic operations.

Reply:
I think it does here. But that's the way `dalek` decided to go: there's no `Keypair` any more (though you can still get the keypair as bytes).